code/game/machinery/newscaster.dm

```diff
@@ -873,6 +878,34 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 		qdel(P)
 
 
+/obj/machinery/newscaster/proc/parsepencode(t, mob/user)
```

Comment:
This is copy-pasted, which is bad because it means that changing the code here means it has to be changed everywhere else too. just make it a global proc. (/proc/parsepencode()) and use that everywhere.